File: QuizForms/Models/Question.cs
using EzCoreKit.AspNetCore.Http;
using Newtonsoft.Json;
using QuizForms.Quiz.Models;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace QuizForms.Models {
    [Table("Question")]
    public class Question : IQuestion<Guid> {
        public Guid Id { get; set; }
        public Guid FormId { get; set; }
        public Guid? ParentId { get; set; }
        public int Order { get; set; }

        [NotMapped]
        public QuestionTypes Type {
            get {
                return (QuestionTypes)Enum.Parse(typeof(QuestionTypes), TypeString);
            }
            set {
                TypeString = value.ToString();
            }
        }

        [Column(name:"type")]
        public string TypeString { get;set; }

        public bool Required { get; set; }
        public string Text { get; set; }
        public Question() {
            Id = Guid.NewGuid();
        }

        [NotMapped]
        [JsonIgnore]
        public IQuestion<Guid> Parent {
            get {
                if (!ParentId.HasValue) return null;
                var db = GetCurrentDbContext();
                return db.Questions.First(x => x.Id.Equals(ParentId));
            }
        }

        [NotMapped]
        private IQuestion<Guid>[] _children = null;

        [NotMapped]
        public IQuestion<Guid>[] Children {
            get {
                if (_children == null) {
                    _children = GetCurrentDbContext().Questions
                        .Where(x => x.ParentId == Id)
                        .OrderBy(x => x.Order).ToArray();
                }
                return _children;
            }
        }

        private static QuizDbContext<Guid, Form, Question, Record, Writed> GetCurrentDbContext() {
            return (QuizDbContext<Guid, Form, Question, Record,Writed>)
                HttpContextFactory.CurrentHttpContext
                .RequestServices.GetService(typeof(QuizDbContext<Guid, Form, Question, Record,Writed>));
        }
    }
}

File: QuizFormsSDK/Question.ts
module QuizForms {
    /**
     * 題目類型
     */
    export enum QuestionTypes {
        /**
         * 多選一
         */
        RadioGroup,
        /**
         * 多選一的選項
         */
        Radio,
        /**
         * 行內多選一(下拉式清單) 以{options}標記
         */
        Dropdown,
        /**
         * 多選勾選框群組
         */
        CheckboxGroup,
        /**
         * 多選勾選框
         */
        Checkbox,
        /** 
         * 五等級多選一
         */ 
        Level,
        /**
         * 文字輸入框
         */
        InputText,
        /** <summary>
         * 數字輸入框
         */
        InputNumber,
        /** <summary>
         * 標題字
         */
        Title,
        /** <summary>
         * 普通內文
         */
        Text,
    }

    /**
     * 題目
     */
    export class Question{
        /**
         * 唯一識別號
         */
        public id: any;

        /**
         * 所屬問卷唯一識別號
         */
        public formId: any;

        /**
         * 問題父節點
         */
        public parentId: any;

        /**
         * 顯示順序
         */
        public order: number;

        /**
         * 類型
         */
        public type: string;

        /**
         * 是否為必填
         */
        public required: boolean;

        /**
         * 內容
         */
        public text: string;

        /**
         * 父節點
         */
        public parent: Question;

        /**
         * 子節點
         */
        public children: Question[];



    }
}